manticore/platforms/linux.py

```diff
@@ -2115,21 +2115,36 @@ def sys_accept4(self, sockfd, addr, addrlen, flags):
         fd = self._open(sock)
         return fd
 
-    def sys_recv(self, sockfd, buf, count, flags):
-        try:
-            sock = self.files[sockfd]
-        except IndexError:
-            return -errno.EINVAL
+    def sys_recv(self, sockfd, buf, count, flags, trace_str="_recv"):
+        data: bytes = bytes()
+        if count != 0:
+            if buf not in self.current.memory:
```

Comment:
You could also consider checking if `buf + count in self.current.memory`

Reply:
Good call. I also see other areas in the code where only `buf not in self.current.memory`. Shall I fix those instances too? There even seems to be a comment about checking `buf+count` (line 1338), but not actually doing it, here:
https://github.com/trailofbits/manticore/blob/7a9533a7a17531a07f0683f472e2c39a68c3271f/manticore/platforms/linux.py#L1334-L1340
Is there a reason for that?

Reply:
We should probably have such check for each byte that is being written OR for the whole slice.

Something similar to `max_exec_size` or whatever name it ends up with, suggested in https://github.com/trailofbits/manticore/pull/1522, might be handy here.

Reply:
Regarding:
> I also see other areas in the code where only buf not in self.current.memory. Shall I fix those instances too? 

That would be awesome. But please, use another PR for that :).

Reply:
Add my change in https://github.com/trailofbits/manticore/pull/1514/commits/b66400090465243abc8a94e4bf1e5b41f76de012